cpp/foxglove/include/foxglove/cloud_sink.hpp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <foxglove/channel.hpp>
+#include <foxglove/context.hpp>
+#include <foxglove/error.hpp>
+#include <foxglove/server.hpp>
+
+struct foxglove_cloud_sink;
+
+namespace foxglove {
+
+/// @brief The callback interface for a Cloud Sink.
+///
+/// These methods are invoked from the client's main poll loop and must be as low-latency as
+/// possible.
+///
+/// @note These callbacks may be invoked concurrently from multiple threads.
+/// You must synchronize access to your mutable internal state or shared resources.
+struct CloudSinkCallbacks {
```

Comment:
Any thoughts on the "Listener"/"Callback" naming now that you've revisited in this context? I recall this being a conscious decision with the WS server, but the inconsistency is slightly unfortunate when documenting concepts.